### DISCUSSION TITLE
How to properly disconnect(terminate client-socket connection) from SERVER SIDE

### DISCUSSION THREAD

**Discussion by PGCodehub** [miguelgrinberg/python-socketio#1059] [Q&A]
HI
  
  I am trying to disconnect a client from the server side using (disconnect(sid, namespace=None, ignore_queue=False))
  
   which disconnects the client , but I have some functions running on the server side (based on sid) with on disconnect event (on server side) which is triggered with NEW SID,  which make running those functions difficult 
  
  
  
  is there a correct way to do a client to disconnect 
   
   Please help me on this , Thank you

**Comment by miguelgrinberg** [comment]
I don't understand what the problem is. The disconnect event handler is invoked with the current sid, I'm not sure what you mean by "new sid" here. Can you please explain better what the problem is?
